clients/deltalake/src/main/java/org/projectnessie/deltalake/ContentKeyUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.deltalake;
+
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.Path;
+import org.projectnessie.model.ContentKey;
+
+/**
+ * Utility class that produces a {@link ContentKey} instance from a Hadoop {@link Path} or a file
+ * path string.
+ */
+public class ContentKeyUtil {
```

Comment:
given that this is really only required for Delta, I decided to put this into the deltalake package